py/mpconfig.h

```diff
@@ -856,6 +856,23 @@ typedef double mp_float_t;
 #define MICROPY_PY_BUILTINS_COMPLEX (MICROPY_PY_BUILTINS_FLOAT)
 #endif
 
+// Float conversion to/from string implementations
+#define MICROPY_FLTCONV_IMPL_BASIC (0)  // smallest code, but inexact
+#define MICROPY_FLTCONV_IMPL_APPROX (1) // slightly bigger, almost perfect
+#define MICROPY_FLTCONV_IMPL_EXACT (2)  // bigger code, but 100% exact repr
+
+#if MICROPY_FLOAT_IMPL == MICROPY_FLOAT_IMPL_FLOAT
+#define MICROPY_FLOAT_DEFAULT_FLTCONV_IMPL (MICROPY_FLTCONV_IMPL_APPROX)
+#elif defined(__SIZEOF_LONG_DOUBLE__) && __SIZEOF_LONG_DOUBLE__ > __SIZEOF_DOUBLE__
+#define MICROPY_FLOAT_DEFAULT_FLTCONV_IMPL (MICROPY_FLTCONV_IMPL_EXACT)
+#else
+#define MICROPY_FLOAT_DEFAULT_FLTCONV_IMPL (MICROPY_FLTCONV_IMPL_APPROX)
+#endif
+
+#ifndef MICROPY_FLTCONV_IMPL
+#define MICROPY_FLTCONV_IMPL (MICROPY_FLOAT_DEFAULT_FLTCONV_IMPL)
```

Comment:
Is this `MICROPY_FLOAT_DEFAULT_FLTCONV_IMPL` needed?  Why not just wrap the above bit of code in `#ifndef MICROPY_FLTCONV_IMPL` and then define `MICROPY_FLTCONV_IMPL` directly?

Also, maybe call it `MICROPY_FLOAT_FORMAT_IMPL`?